src/main/java/org/jenkinsci/plugins/ghprb/extensions/build/GhprbCancelBuildsOnUpdate.java

```diff
@@ -39,15 +39,15 @@ public Boolean getOverrideGlobal() {
         return overrideGlobal == null ? false : overrideGlobal;
     }
 
-    private void cancelCurrentBuilds(AbstractProject<?, ?> project,
+    private void cancelCurrentBuilds(Job<?, ?> project,
                                      Integer prId) {
         if (getOverrideGlobal()) {
             return;
         }
         
         logger.log(Level.FINER, "New build scheduled for " + project.getName() + " on PR # " + prId + ", checking for queued items to cancel.");
         Queue queue = Jenkins.getInstance().getQueue();
-        for (Queue.Item item : queue.getItems(project)) {
+        for (Queue.Item item : queue.getItems()) {
```

Comment:
@mmitche That's my understanding as well. `queue.getItems()` returns projects from the Jenkins' queue's `blockedProjects`, `buildables`, `pending` and `waitingList` properties. This call would cancel all of those pending projects, even if it doesn't match this project.

My solution was to replace

`
for (Queue.Item item : queue.getItems()) {
`

with:

```
        Queue.Item queueItem = project.getQueueItem();
    
             if (queueItem != null) {
```

Although, this fix assumes there's only 1 item in the waiting queue to potentially cancel.